src/spawn_sys.rs

```diff
@@ -0,0 +1,450 @@
+use crate::components::*;
+use crate::events::{SpawnGuidedMissileEvent, SpawnMineEvent};
+use crate::ship_parts::*;
+use crate::ships::*;
+use bevy::prelude::*;
+use bevy::window::PrimaryWindow;
+use rand::prelude::*;
+use std::f32::consts::PI;
+
+pub fn spawn_player_system(
+    mut commands: Commands,
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    asset_server: Res<AssetServer>,
+) {
+    let window: &Window = window_query.get_single().unwrap();
+    commands.spawn((
+        SpriteBundle {
+            transform: Transform::from_xyz(window.width() / 2.0, window.height() / 2.0, 0.0)
+                .with_scale(GLOBAL_RESCALE_V),
+            texture: asset_server.load("sprites/ships/playerShip1_blue.png"),
+            ..default()
+        },
+        Player {},
+        Ship {
+            turn_speed: f32::to_radians(1.25),
+            primary_weapon: load_basic_torpedo(),
+            secondary_weapon: load_basic_laser(),
+            tertiary_weapon: load_basic_cannon(),
+        },
+        Velocity {
+            velocity: Vec3 {
+                x: 0.0,
+                y: 0.0,
+                z: 0.0,
+            },
+        },
+        load_basic_thruster(),
+        Clipping {
+            cd_timer: Timer::from_seconds(0.0, TimerMode::Once),
+        },
+        Drag {
+            dampening_factor: 0.995,
+        },
+        CollisionBox {
+            shape: Shape::Circle,
+            width_radius: 38.0 * GLOBAL_RESCALE_C,
+            height: 38.0 * GLOBAL_RESCALE_C,
+        },
+        Health { value: 500.0 },
+        Mass { value: 100000.0 },
+        EntityType::Ship,
+    ));
+}
+
+pub fn read_script_system(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    level_script: Res<LevelScript>,
+    mut script_line: ResMut<CurrentScriptLine>,
+    mut script_timer: ResMut<ScriptTimer>,
+    q_window: Query<&Window, With<PrimaryWindow>>,
+) {
+    if script_timer.delay.finished() {
+        let mut rng = thread_rng();
+        let window = q_window.get_single().unwrap();
+        for (i, line) in level_script.txt.as_str().trim().lines().enumerate() {
+            // Only want to read from the current line_num onwards.
+            if i < script_line.line_num {
```

Comment:
This will get progressively more inefficient the longer your script gets. An easy way to avoid that would be to remember/store the offset in the string where you left off last time, instead of the line number. You can then jump straight to the relevant position in the string: `level_script.txt[offset..].lines().enumerate()` to not have to scan the previous part again.